### PR TITLE
include common directory in tests on path changes

### PR DESCRIPTION
**Pull request by Gregory-Pereira**
Fixing tests, will also use this PR to fix any code changes that correspond to failing tests. 

Broken out from #258.

Changes:

- add a check on jobs in the standard development workflows (all apps, model_servers, model_image_build_push) to not run if there is a `hold-tests` label, I thought this could save us some resources
- fix `codegen`'s `Download model` step due changing make target name
- fix `chromedriver` and `chrome`  downloads failling due to timeout
    - move binaries to `/recipes/common/bin`
- swapped `install` make target in both `recipes/common` and each recipe to be a [double colon rule](https://www.gnu.org/software/make/manual/html_node/Double_002dColon.html#:~:text=4.13%20Double%2DColon%20Rules&text=When%20a%20target%20appears%20in,any%20prerequisites%20of%20that%20rule.) rather than a single colon rule, which allows us to run common and local install commands together

### CODE REVIEW

**Comment by Gregory-Pereira** [comment]
Tests pass, going to add the `hold-tests` label and push an empty commit to see if it works.

**Comment by Gregory-Pereira** [comment]
Can confirm that the labels to skip tests work. Removing `hold-tests` label

File: .github/workflows/chatbot.yaml [hunk 1]
```diff
@@ -22,6 +24,7 @@ env:
 
 jobs:
   build-and-push-image:
+    if: "!contains(github.event.pull_request.labels.*.name, 'hold-tests')"
```
Comment (lmilbaum) [inline .github/workflows/chatbot.yaml:27]: Not sure folks would remember to use this trick

File: recipes/common/Makefile.common [hunk 2]
```diff
@@ -129,9 +129,9 @@ bootc-image-builder: bootc
 install-chromedriver:
 	@if [[ -z "$(LOCAL_CHROMEDRIVER_EXISTS)" ]]; then \
 		if [[ -n "$(UNZIP_EXISTS)" ]]; then \
-			curl -sLO $(CHROMEDRIVER_MIRROR)/$(CHROMEDRIVER_VERSION)/$(CHROMEDRIVER_DOWNLOAD_PATH); \
+			curl --max-time 300 --connect-timeout 60 -sLO $(CHROMEDRIVER_MIRROR)/$(CHROMEDRIVER_VERSION)/$(CHROMEDRIVER_DOWNLOAD_PATH); \
```
Comment (lmilbaum) [inline recipes/common/Makefile.common:132]: Nice. I was looking into a retry option. That is a better option to solve those flaky results.

**Review by lmilbaum — APPROVED** [review]
LGTM

File: .github/workflows/chatbot.yaml (hunk 1 above)
Comment (Gregory-Pereira) [reply .github/workflows/chatbot.yaml:27]: I was going to have a discussion with the team about it.  I know I would use it because sometime I just want to see the diff for review purposes and dont want to waist the resources. If your worried about visibility maybe we could tag every PR with `hold-tests` to begin with and then use a `/ok-to-test` similar to prow?
